src/models/message.ts

```diff
@@ -239,11 +206,11 @@ export class MessageModel extends DocumentDbModel<
    * @param fiscalCode The fiscal code of the recipient
    * @param messageId The ID of the message
    */
-  public async findMessageForRecipient(
+  public findMessageForRecipient(
     fiscalCode: FiscalCode,
     messageId: string
-  ): Promise<Either<DocumentDb.QueryError, Option<RetrievedMessage>>> {
-    const errorOrMaybeMessage = await this.find(messageId, fiscalCode);
+  ): TaskEither<CosmosErrors, Option<RetrievedMessage>> {
+    const errorOrMaybeMessage = super.find(messageId, fiscalCode);
 
     return errorOrMaybeMessage.map(maybeMessage =>
```

Comment:
why do we need this filter here ? isn't the fiscalCode already filtered by the query?

Reply:
I asked myself the same question, I believe it's there as a "double check to be sure", not a big deal though